amazon/ion/ioncmodule.c

```diff
@@ -193,11 +195,11 @@ static iERR c_string_from_py(PyObject* str, char** out, Py_ssize_t* len_out) {
  */
 static iERR ion_string_from_py(PyObject* str, ION_STRING* out) {
     iENTER;
-    char* c_str = NULL;
+    const char* c_str = NULL;
     Py_ssize_t c_str_len;
     IONCHECK(c_string_from_py(str, &c_str, &c_str_len));
     ION_STRING_INIT(out);
-    ion_string_assign_cstr(out, c_str, c_str_len);
+    ion_string_assign_cstr(out, (char *)c_str, c_str_len);
```

Comment:
I updated `c_string_from_py` to be `const char **`. Ion-C's string functions drop the const, so I've isolated the impact to the ion-c calls themselves by dropping the const via typecast and adding them everywhere else.